### DISCUSSION TITLE
Use cliphist but the dmenu is blank

### DISCUSSION THREAD

**Discussion by DzmingLi** [davatorium/rofi#1957] [Q&A]
My wayland compositor is niri.
I add MOD+V { spawn "sh" "-c" "cliphist list | rofi -dmenu | cliphist decode | wl-copy";} in the config.kdl.
I type "cliphist list | rofi -dmenu |cliphist decode | wl-copy" , it works. But press MOD+V the rofi dmenu shows, but it is blank. (I am sure that the cliphist is not blank as I have typed it  in the shell before)

**Comment by DaveDavenport** [comment]
If things behave differently on terminal then from a keybinding check the environment it gets in the keybinding. 

If it works on the terminal, then rofi seems to be working correctly. If you get an empty list I suspect chiplist is not working well from keybinding. I would check this first.

I see you use wayland? do you use this rofi with X emulation or @lbonn port to wayland?

**Reply by lbonn** [reply]
I agree, it does not look rofi-related.

You could try to log errors from `cliphist list`, see if you see something interesting

```
MOD+V { spawn "sh" "-c" "cliphist list > /tmp/cliphist.log 2>&1" }
```

And more radical:

```
MOD+V { spawn "sh" "-c" "strace cliphist list > /tmp/cliphist.log 2>&1" }
```
